sdk/src/account_utils.rs

```diff
@@ -37,10 +39,32 @@ where
     }
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum SystemAccountKind {
+    System,
+    Nonce,
+}
+
+pub fn account_is_system(account: &Account) -> Option<SystemAccountKind> {
```

Comment:
I'm curious about how this is used.  Where is this needed?

Reply:
It just resolves our system account flavor. I used it in all of the instruction guards and rent-exempt checks